### ISSUE TITLE
“Copy Files” toggle is ignored when duplicating a page

### ISSUE THREAD

**Issue by linusrogge** [k-next/kirby#6885]
## Description
When duplicating a page with images, I have to option to set a toggle whether to copy these files or not. But no matter what option it is set to, the files are _always copied_.

**Expected behavior**  
When the toggle is set to `false`, no files should appear in the duplicated page.

This looks like the problem. Shouldn’t it be `if ($files !== false) { …`?

https://github.com/getkirby/kirby/blob/94cc37ee7c3004ebb4950a53f14e1329ed4d28d3/src/Cms/PageActions.php#L470-L479

## To reproduce

1. Create a page and upload an image
2. Click on “Duplicate” and leave “Copy Files” toggle on false
3. Click duplicate
4. See copied files

## Your setup
**Kirby Version**  
4.5.0 (but has been around before)

**Your system (please complete the following information)**
 - Device: MacBook Pro 
 - OS: macOS Sequoia, Version 15.2 (24C101)
 - Browser: Chrome

**Comment by afbora** [comment]
Hello @linusrogge I've tried on starterkit 4.5.0 and it's working great. Do you have plugins?

Also I just wonder that copy and don't copy actions works for you with `$files !== false` condition?

**Comment by linusrogge** [comment]
So, I have done a bit of testing and things are weirder than I thought. 

I tested around with the Starterkit as well and it does work mostly correctly there indeed—however, and this is what I also discovered on my sites: Kirby, when the toggle is set to false, does _not_ copy the files. But in the `txt` file of the duplicated page, the UUID from the file from the original page is still included. Can you follow me?

### Example

1. I duplicate the “Mountains” album in the Starterkit
2. I leave the “Copy files” toggle on “off”
3. I click “Duplicate”
4. The duplicate opens, and in the “Cover” field, there is still an image visible

<img width="446" alt="Image" src="https://github.com/user-attachments/assets/85ea25f8-95ab-425a-847b-682be2aa25cf" />

However, when I click it, it opens the file from the original page (notice the breadcrumbs at the top: Image above is “Mountains Copy”, image below just says “Mountains” which is the original)

<img width="634" alt="Image" src="https://github.com/user-attachments/assets/99edb01d-33a8-427b-ada2-839355b42fcd" />

And also in the `_drafts/mountains-copy/album.txt` file, there is already a value set for “Cover” although I haven’t done anything other than duplicate the page:

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/6d98375c-666e-4c95-99ff-ccb231c9b86d" />

Then again: the `_drafts/moutains-copy` folder is empty (except for the `.txt` file)  and, as to no surprise, when I open the `probably-photoshopped.jpg.txt` from the original page, it has the UUID that is also linked in the duplicated page. 

Now, when I click the “Select” button in the “Cover” field, the dialog is empty

<img width="892" alt="Image" src="https://github.com/user-attachments/assets/17cfa3aa-1a16-4dee-a474-8a5ede7aa9eb" />

---

In conclusion: there’s something off with the duplicate logic which does _not actually copy the files_, but simply persists the value of the field in the newly created `.txt` file, wrongly showing the image from the original page in its duplicate.

I hope this makes sense, otherwise I am happy to record a loom going through the whole process to showcase it better.

**Comment by distantnative** [comment]
This behavior will be changed with Kirby 5 (already present in the beta): https://github.com/getkirby/kirby/pull/6567

With Kirby 4, duplicating a page would keep all UUID references as they are. So no matter if duplicating files or not, the old UUID references would always point to the files of the old page.

With Kirby 5, we will differentiate between UUID references to files within the page and outside of the page. Those to files outside of the page will always stay the same when duplicating a page. But those to files form within the page (or subpages) will be treated differently: when duplicating files as well, we will replace the UUID references to the files within the old page with those new duplicated files within the new page. When not copying over the files, we will delete any UUID referencing one of those files that didn't get copied.

I think this will solve your expected outcome.

**Comment by linusrogge** [comment]
Glad to hear this will be part of Kirby 5. While trying this behaviour using beta-1, I had a different issue where leaving the toggle set to false would not let me duplicate the page at all:

![image](https://github.com/user-attachments/assets/389c5648-1938-445d-86d8-fbf002a9ebfa)
